CMakeCommon.cmake

```diff
@@ -99,8 +99,103 @@ function (LinkGSLibrariesToProject target acVersion devKitDir)
+endfunction ()
+
+function (GenerateAddOnVersionInfo target)
+    get_target_property (devKitDir "${target}" GSdevKitDir)
+    get_target_property (addOnName "${target}" GSaddOnName)
+    get_target_property (acVersion "${target}" GSacVersion)
+    if (NOT DEFINED devKitDir OR NOT DEFINED addOnName OR NOT DEFINED acVersion)
+        message (FATAL_ERROR "Target '${target}' was not created with GenerateAddOnProject.")
+    endif ()
+
+    cmake_parse_arguments (PARSE_ARGV 1 vers "" VERSION "")
+    if (DEFINED vers_KEYWORDS_MISSING_VALUES AND "VERSION" IN_LIST vers_KEYWORDS_MISSING_VALUES)
+        message (FATAL_ERROR "\
+'VERSION' argument is missing its value. Please make sure you replaced the placeholder in CMakeLists.txt:
+
+    GenerateAddOnVersionInfo (CMakeTarget VERSION #[[version goes here]])
+
+Replace the '#[[version goes here]]' comment with a version number that you wish see in the file's properties and bug reports.
+Accepted version number formats are:
+
+    123
+    1.23
+    1.2.3")
+    endif ()
+
+    ParseVersion ("${vers_VERSION}" vers)
+    if (NOT DEFINED vers)
+        message (FATAL_ERROR "'${vers_VERSION}' does not follow the '1.2.3' version format, where the 2nd and 3rd components are optional.")
+    endif ()
+
+    set (company "GRAPHISOFT SE")
+    string (TIMESTAMP copyright "Copyright © ${company}, 1984-%Y")
+
+    if (WIN32)
+        # FIXME(HVA): include GS build num in Windows release as well
+        list (APPEND vers "${acVersion}")
+        list (JOIN vers , version_comma)
+        list (JOIN vers . version)
+
+        set (out "${CMAKE_CURRENT_BINARY_DIR}/${target}-VersionInfo.rc")
+        configure_file ("${CMAKE_CURRENT_FUNCTION_LIST_DIR}/VersionInfo.rc.in" "${out}" @ONLY)
+        target_sources ("${target}" PRIVATE "${out}")
+    else ()
+        # BE on the safe side; load the info from an existing framework
+        file (READ "${devKitDir}/Frameworks/GSRoot.framework/Versions/A/Resources/Info.plist" plist_content NEWLINE_CONSUME)
+        string (REGEX MATCH "GSBuildNum[^0-9]+([0-9]+)" unused "${plist_content}")
+        set (gsBuildNum "${CMAKE_MATCH_1}")
+        string (REGEX MATCH "LSMinimumSystemVersion[^0-9]+([0-9.]+)" unused "${plist_content}")
+        set (lsMinimumSystemVersion "${CMAKE_MATCH_1}")
+
+        list (APPEND vers "${gsBuildNum}")
+        list (JOIN vers . version)
+
+        string (TOLOWER "${addOnName}" lowerAddOnName)
+        string (REGEX REPLACE "[ _]" "-" addOnNameIdentifier "${lowerAddOnName}")
+
+        set (MACOSX_BUNDLE_EXECUTABLE_NAME "${addOnName}")
+        set (MACOSX_BUNDLE_INFO_STRING "${addOnName}")
+        set (MACOSX_BUNDLE_GUI_IDENTIFIER "com.graphisoft.${addOnNameIdentifier}")
+        set (MACOSX_BUNDLE_LONG_VERSION_STRING "${copyright}")
+        set (MACOSX_BUNDLE_BUNDLE_NAME "${addOnName}")
+        set (MACOSX_BUNDLE_SHORT_VERSION_STRING "${version}.${gsBuildNum}")
+        set (MACOSX_BUNDLE_BUNDLE_VERSION "${version}.${gsBuildNum}")
```

Comment:
The AC version used to be included in the version number, but now it isn't.

Does it need to be included? If yes, can it be included in the 4th version component via multiplication? Example:

```
AC_VERSION: 28
GSBuildNum: 1234
VERSION argument: 1.2.3
4th component: AC_VERSION * 100000 + GSBuildNum = 2801234
Final version: VERSION argument . 4th component = 1.2.3.2801234
```

This way the embedded version number follows semantic versioning in a way and both the AC version and the build number can be retrieved.